mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -156,8 +160,16 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 	b.kubeHandler = &kubeHandler
 
 	if !found {
-		if err := initMultiClusterSecretController(b, path, env); err != nil {
-			return nil, fmt.Errorf("could not create remote controllers: %v", err)
+		// only init a single watcher for multicluster secrets for a given Builder.
+		// if init fails, try again on next Build().
+		// This should prevent an ever-growing number of watchers for a Builder.
+		var initErr error
+		b.multiClusterWatcherInit.Do(func() {
+			initErr = initMultiClusterSecretController(b, path, env)
+		})
+		if initErr != nil {
+			b.multiClusterWatcherInit = sync.Once{}
```

Comment:
FYI, I don't think this is safe. Per https://golang.org/pkg/sync/, a sync.Once should not be copied, but that's what this line is doing.

The better solution would be to use a simple mutex and check for the initialisation already being done under that.